pkg/ebpf/controlplane/controller.go

```diff
@@ -172,6 +178,10 @@ func (ctrl *Controller) processSignal(signal *signal) error {
 		return ctrl.procTreeExitProcessor(signal.args)
 	case events.SignalHeartbeat:
 		heartbeat.SendPulse()
+	default:
+		// Handle remaining signals via registered external signal handlers
+		// This allows extensions to process signals via their own handlers
+		return ctrl.processExternalSignal(signal)
```

Comment:
I think the name `processExternalSignal` doesn't accurately reflect what it actually does. Since it's not used elsewhere, you could try removing the helper and just check for the signal handler (and call it) directly here. Also, updating the comment would help clarify its purpose.